### ISSUE TITLE
extract returning RAW result when returned in Hash

### ISSUE THREAD

**Issue by leviwilson** [neo4jrb/neo4j#979]
## Description
I came across an issue that I mentioned to @cheerfulstoic in the Slack channel that when I was doing an `extract` like so:

```ruby
entity.sentences(:s, :r, rel_length: 2..4)
  .pluck('s, extract(rel in r | {from: startNode(rel), to: endNode(rel)})')
```
It would return the RAW cypher vs. recognizing that it should be rolled into an `ActiveNode` record.

Using an array does the appropriate action:

```ruby
entity.sentences(:s, :r, rel_length: 2..4)
  .pluck('s, extract(rel in r | [startNode(rel), endNode(rel)])')
```

**Comment by subvertallchris** [comment]
I think there's an easy fix for this. I'm going to give it a shot tonight.

**Comment by cheerfulstoic** [comment]
There's some code in my branch here:

https://github.com/neo4jrb/neo4j-core/blob/cypher_session_reboot/lib/neo4j/core/cypher_session/responses/http.rb#L44

You could probably steal it for this.  I think it should make a pretty good balance of performance and conservative identification.

**Comment by cheerfulstoic** [comment]
Looking into this now...

**Comment by cheerfulstoic** [comment]
Fixed and released as 5.1.6 (confirmed with @leviwilson on Slack)

**Comment by leviwilson** [comment]
Fixed in https://github.com/neo4jrb/neo4j-core/commit/1bc1dcc2dc8d7a9c071da9181504f27980d1785f
